src/software/ai/evaluation/q_learning/attacker_mdp_feature_extractor.cpp

```diff
@@ -0,0 +1,43 @@
+#include "software/ai/evaluation/q_learning/attacker_mdp_feature_extractor.h"
+
+#include <math.h>
+
+#include "software/ai/evaluation/possession.h"
+
+AttackerMdpFeatureExtractor::AttackerMdpFeatureExtractor()
+    : FeatureExtractor({ballXPositionFeature, bestPassRatingFeature, bestShotOpenAngleFeature})
+{
+}
+
+double AttackerMdpFeatureExtractor::ballXPositionFeature(const AttackerMdpState& state)
+{
+    const double x_pos = state.world_ptr->ball()
```

Comment:
why do we estimate future state here and not just use the real x value? Is it to incentivize forward progress rather than staying at an already high x coordinate?

Reply:
This is leftover from when I tried making the attacker become the receiver after kicking the ball. I will change it back to using the actual ball position